graphql_compiler/macros/macro_edge/helpers.py

```diff
@@ -139,7 +173,16 @@ def _replace_tag_names_at_current_node(name_change_map, ast):
 
 
 def replace_tag_names(name_change_map, ast):
-    """Replace tag names that are already in use in the whole AST."""
+    """Return a new ast with tag names replaced according to the name_change_map.
```

Comment:
Everywhere where you depend on the exact same object being returned in the "no changes" scenario, please make sure to call that out in the docstring. Otherwise, it's not a postcondition and the code that depends on that behavior is technically wrong.

Reply:
It is not a postcondition. I don't rely on it.

Reply:
Is it misleading to say `Return a new ast...`, but in the case where nothing needs to be done, return the same one? Is `Return an ast ...` clearer?

Reply:
I'd just append a new sentence, something like `If no tag names were replaced, the same AST object that was provided as input is returned.` What do you think?